src/web/mjs/connectors/Comikey.mjs

```diff
@@ -0,0 +1,66 @@
+import Connector from '../engine/Connector.mjs';
+
+export default class ComiKey extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'comikey';
+        super.label = 'Comikey';
+        this.tags = ['manga', 'webtoon', 'english'];
+        this.url = 'https://comikey.com';
+        this.chapterUrl = 'https://relay-us.epub.rocks/consumer/COMIKEY/series/';
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this._getMangasFromPage(page);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+            mangas.length == 25 ? run = true : run = false;
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        let uri = new URL('/sapi/comics', this.url);
+        uri.searchParams.set('page', page);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        return data.results.map(manga => {
+            return {
+                id: manga.id + ',' + manga.e4pid,
+                title: manga.name
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        const uri = new URL(manga.id.split(',')[1] + '/content?clientid=dylMNK5a32of', this.chapterUrl);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        return data.data.episodes.map(chapter => {
+            let name = chapter.name;
+            let title = '';
+            if (name.length > 0) {
+                title = name.shift().name;
+            }
+            title = title ? ' - ' + title : '';
+            return {
+                id: chapter.id,
+                title: chapter.number + title,
+                language: chapter.language
+            };
+        });
+    }
+
+    async _getPages(chapter) {
+        let uri = new URL('sapi/comics/' + chapter.manga.id.split(',')[0] + '/read', this.url);
```

Comment:
Is this the correct API call?
I got the image links from an URL such as
https://gate.epub.rocks/delivery/COMIKEY-EPI-oKA33D_web/sK8drehLxwWwYCAnfaEbQ.json~divina?token=8c5ab42efc9595fd25b0bace7c6854ee63cd89e5ca907f9bc94f94abe18b2589&expires=617ea15f&scope=%2Fdelivery%2FCOMIKEY-EPI-oKA33D_web%2F&anonymous

Reply:
There are 2 api calls in this request. First one is like this: https://comikey.com/sapi/comics/35/read?format=json&content=EPI-qD0ZEo which returns a json containg link to the pages one you mentioned above.